src/dev/flang/fe/SourceModule.java

```diff
@@ -1283,6 +1294,103 @@ else if ((t1.isChoice()
             AstErrors.constraintMustNotBeGenericArgument(f);
           }
       }
+    checkLegalTypeVisibility(f);
+    checkResultTypeVisibility(f);
+    checkArgTypesVisibility(f);
+    checkPreconditionVisibility(f);
+    checkOuterVisibility(f);
+  }
+
+
+  /**
+   * Check that type of outer feature of `f` isn't less visible than `f` itself.
+   * @param f
+   */
+  private void checkOuterVisibility(Feature f)
+  {
+    if (f.outer() != null && f.outer().visibility().typeVisibility().ordinal() < f.visibility().ordinal())
```

Comment:
I think we will have to drop this restriction, as discussed, a private feature should be able to define a public `as_String`.

Reply:
I dropped this check.